contrib/stack/topsStack/Stack.py

```diff
@@ -13,6 +13,19 @@
 maxNodes = 72
 
 
+
+def writeBash(self, configName, line_cnt):
```

Comment:
I would recommend the following changes:
1. move this function inside `config` class since it uses `self.runf`
2. add `numProcess` as an argument with the default value of 1, and pass `self.numProcess` only for steps/functions applicable. This will allow us to expand it easily in the futurer.
3. maybe renamed to `write_wrapper_config2run_file()`?

Reply:
Thanks @yunjunz I have tried moving it inside the config class but I get the following issue when running

```
Traceback (most recent call last):
  File "/mnt/d/dev/isce2/contrib/stack/topsStack/stackSentinel.py", line 765, in <module>
    main()
  File "/mnt/d/dev/isce2/contrib/stack/topsStack/stackSentinel.py", line 748, in main
    interferogramStack(inps, acquisitionDates, stackReferenceDate, secondaryDates, safe_dict, pairs, updateStack)
  File "/mnt/d/dev/isce2/contrib/stack/topsStack/stackSentinel.py", line 551, in interferogramStack
    i = slcStack(inps, acquisitionDates, stackReferenceDate, secondaryDates, safe_dict, updateStack)
  File "/mnt/d/dev/isce2/contrib/stack/topsStack/stackSentinel.py", line 479, in slcStack
    runObj.pairs_misregistration(acquisitionDates, safe_dict)
  File "/mnt/d/dev/isce2/contrib/stack/topsStack/Stack.py", line 498, in pairs_misregistration
    line_cnt = configObj.writeBash(self, configName, line_cnt)
TypeError: writeBash() takes 3 positional arguments but 4 were given
```

I intend to take all of your recommendations on board. That said if I am too slow I am very happy if you choose to implement it.

The renaming suggestion made me chuckle as I had to choose between the `camelBack` or `under_score` conventions throughout the collaborative code. I chose wrong!

Reply:
Note: I assume your thinking is to use the existing config constructor.

Reply:
> Note: I assume your thinking is to use the existing config constructor.
Yes, so we won't need the `self` argument while calling this function.

For the renaming, I see both `camelBack` and `under_score` in the code, so whichever style you choose would be fine!

Reply:
See new commit. 

I think your comment on the issue https://github.com/isce-framework/isce2/issues/223 was to use this function wherever possible? This made sense to me.

Keeping track of line counts doesn't thrill me but it seems to work. Happy to take another look at this. 

From the command line arguments to `stackSentinel.py` I tested
* Not including `--num-process`
* `--num-process 1` (give the same result as withholding it)
* `--num-process 6`

https://mintpy.readthedocs.io/en/latest/#5_contributing was right :)

Reply:
Looks great! My last minor comment would be to use 4 whitespace indentation in `write_wrapper_config2run_file()`.